### PR TITLE
Signature fix of Solr's backend terms method

### PR DESCRIPTION
**Pull request by ThoWagen**
We have the case that we want to use the 'TermsCommand'. However, the query parameter "terms.lower=" causes an empty search result. So we need to be able to set 'start' to null.

This PR makes the arguments of the 'TermsCommand' nullable and sets proper type definitions for the 'terms' method of the Solr backend.

### CODE REVIEW

File: module/VuFindSearch/src/VuFindSearch/Command/TermsCommand.php [hunk 1]
```diff
@@ -111,7 +111,7 @@ public function getArguments(): array
      *
      * @return string
      */
-    public function getField(): string
+    public function getField(): ?string
```
Comment (demiankatz) [inline module/VuFindSearch/src/VuFindSearch/Command/TermsCommand.php]: The doc comments above the getters should be adjusted to have consistent types.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @ThoWagen -- just noticed one small omission.

There's also a question of whether we should release this in 10.1 or 11.0, since it's technically a backward-breaking change due to the signature changes. However, since it could be argued to be a bug fix, maybe it's justifiable to release sooner with a changelog note, especially since I don't think this is likely to be frequently customized.

File: module/VuFindSearch/src/VuFindSearch/Command/TermsCommand.php (hunk 1 above)
Comment (ThoWagen) [reply module/VuFindSearch/src/VuFindSearch/Command/TermsCommand.php]: fixed

**Comment by ThoWagen** [comment]
I am fine both 10.1 and 11.0. 11.0 is probably better to be on the safe side.

**Comment by demiankatz** [comment]
> I am fine both 10.1 and 11.0. 11.0 is probably better to be on the safe side.

Thanks! I've put this into the 11.0 milestone and will merge it once we begin a dev-11.0 branch. If you'd like to get the fix in place sooner, it might make sense to split this in half and merge the command changes to 10.0.1 or 10.1 as a bug fix (since it appears to be an oversight that the signatures of the command are incompatible with the behavior of the backend, and command overrides are even less likely than backend overrides) and then improve the backend signature as an improvement in 11.0. But if you don't think that's worth the effort, we can just let the whole thing go into 11.0.

**Comment by ThoWagen** [comment]
That is a good idea. I create #3891 for the fix in the command. Thanks!

**Review by demiankatz — APPROVED** [review]
Now that the dev-11.0 branch is underway, I'm going to merge this. Thanks again, @ThoWagen!

**Pull request merged**